ocrd_utils/ocrd_utils/os.py

```diff
@@ -106,24 +102,24 @@ def list_all_resources(executable, is_dir=False):
     for parent in candidates:
         if parent.is_dir():
             candidates += parent.iterdir()
-    if is_dir:
-        candidates = [x for x in candidates if x.is_dir()]
-    else:
-        candidates = [x for x in candidates if not x.is_dir()]
     return [str(x) for x in candidates]
 
-def are_processor_resources_directories(executable, ocrd_tool=None):
+def get_processor_resource_types(executable, ocrd_tool=None):
     """
-    XXX if a processor has at least one parameter with
-    `content-type == text/directory`, assume ALL file parameters
-    point to directories, not files
+    Determine whether a processor has resource parameters that represent
+    directories (``has_dirs``), files (``has_files``) or neither.
+
+    Returns a pair ``(has_dir, has_files)``
     """
     if not ocrd_tool:
+        # if the processor in question is not installed, assume both files and directories
         if not which(executable):
-            return False
+            return (True, True)
         result = run([executable, '--dump-json'], stdout=PIPE, check=True, universal_newlines=True)
         ocrd_tool = loads(result.stdout)
-    return next((True for p in ocrd_tool['parameters'].values() if 'content-type' in p and p['content-type'] == 'text/directory'), False)
+    has_dirs = next((True for p in ocrd_tool['parameters'].values() if 'content-type' in p and p['content-type'] == 'text/directory'), False)
+    has_files = next((True for p in ocrd_tool['parameters'].values() if 'content-type' in p and p['content-type'] != 'text/directory'), False)
```

Comment:
Perhaps we should reformulate this so that lazy processors that do not use `content-type` for its file resources at all are assumed to be file-only:
```suggestion
    has_dirs = next((True for p in ocrd_tool['parameters'].values() if p.get('content-type', None) == 'text/directory'), False)
    has_files = next((True for p in ocrd_tool['parameters'].values() if p.get('content-type', None) != 'text/directory'), False)
```

Reply:
No, wait that does not work, because then `has_files` is basically always true because the absence of a `content-type` on **any** parameters makes it true.

Reply:
86c9e290